### ISSUE TITLE
Better print out if bndrun verification fails

### ISSUE THREAD

**Issue by juergen-albert** [bndtools/bnd#5985]
If the resolver plugin is set to verify and fails it prints out the list of actual `-runbundles`  and the list of calculated bundles. They are two separate lines and finding the different is quite hard. 

We should find a better printout two compare the two if possible.

**Comment by pkriens** [comment]
Are you going to provide a PR?

**Comment by juergen-albert** [comment]
I should have provided some more details. The maven resolver plugin can run in a verify mode, where it resolves and looks if the result is different from the actual `runbundles`. The result can look like this:

```
Error:  Error   : Fail on changes set to true (--xchange,-x) and there are changes
Error:  Error   :    Existing runbundles   [com.fasterxml.jackson.core.jackson-annotations;version='[2.16.1,2.16.2)', com.fasterxml.jackson.core.jackson-core;version='[2.16.1,2.16.2)', com.fasterxml.jackson.core.jackson-databind;version='[2.16.1,2.16.2)', io.dropwizard.metrics.core;version='[4.2.19,4.2.20)', junit-jupiter-api;version='[5.9.0,5.9.1)', junit-jupiter-engine;version='[5.9.0,5.9.1)', junit-jupiter-params;version='[5.9.0,5.9.1)', junit-platform-commons;version='[1.9.0,1.9.1)', junit-platform-engine;version='[1.9.0,1.9.1)', junit-platform-launcher;version='[1.9.0,1.9.1)', org.apache.aries.component-dsl.component-dsl;version='[1.2.2,1.2.3)', org.apache.aries.typedevent.bus;version='[0.0.2,0.0.3)', org.apache.commons.commons-csv;version='[1.9.0,1.9.1)', org.apache.felix.configadmin;version='[1.9.24,1.9.25)', org.apache.felix.http.servlet-api;version='[2.1.0,2.1.1)', org.apache.felix.scr;version='[2.2.2,2.2.3)', org.eclipse.emf.common;version='[2.28.0,2.28.1)', org.eclipse.emf.ecore;version='[2.33.0,2.33.1)', org.eclipse.emf.ecore.xmi;version='[2.18.0,2.18.1)', org.eclipse.jetty.alpn.client;version='[11.0.13,11.0.14)', org.eclipse.jetty.client;version='[11.0.13,11.0.14)', org.eclipse.jetty.http;version='[11.0.13,11.0.14)', org.eclipse.jetty.io;version='[11.0.13,11.0.14)', org.eclipse.jetty.security;version='[11.0.13,11.0.14)', org.eclipse.jetty.server;version='[11.0.13,11.0.14)', org.eclipse.jetty.util;version='[11.0.13,11.0.14)', org.eclipse.sensinact.gateway.core.annotation;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.api;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.emf-api;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.geo-json;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.impl;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.models.metadata;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.models.provider;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.device-factory.device-factory-core;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.device-factory.parser-csv;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.http.http-device-factory;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.http.http-device-factory-tests;version='[0.0.2,0.0.3)', org.gecko.emf.osgi.component;version='[5.0.0,5.0.1)', org.opentest4j;version='[1.2.0,1.2.1)', org.osgi.service.cm;version='[1.6.1,1.6.2)', org.osgi.service.component;version='[1.5.0,1.5.1)', org.osgi.service.typedevent;version='[1.0.0,1.0.1)', org.osgi.test.common;version='[1.2.1,1.2.2)', org.osgi.test.junit5;version='[1.2.1,1.2.2)', org.osgi.util.converter;version='[1.0.9,1.0.10)', org.osgi.util.function;version='[1.1.0,1.1.1)', org.osgi.util.promise;version='[1.3.0,1.3.1)', org.osgi.util.pushstream;version='[1.0.2,1.0.3)', slf4j.api;version='[1.7.36,1.7.37)', slf4j.simple;version='[1.7.36,1.7.37)']
Error:  Error   :    Calculated runbundles [com.fasterxml.jackson.core.jackson-annotations;version='[2.16.1,2.16.2)', com.fasterxml.jackson.core.jackson-core;version='[2.16.1,2.16.2)', com.fasterxml.jackson.core.jackson-databind;version='[2.16.1,2.16.2)', io.dropwizard.metrics.core;version='[4.2.19,4.2.20)', junit-jupiter-api;version='[5.9.0,5.9.1)', junit-jupiter-engine;version='[5.9.0,5.9.1)', junit-jupiter-params;version='[5.9.0,5.9.1)', junit-platform-commons;version='[1.9.0,1.9.1)', junit-platform-engine;version='[1.9.0,1.9.1)', junit-platform-launcher;version='[1.9.0,1.9.1)', org.apache.aries.component-dsl.component-dsl;version='[1.2.2,1.2.3)', org.apache.aries.typedevent.bus;version='[0.0.2,0.0.3)', org.apache.commons.commons-csv;version='[1.9.0,1.9.1)', org.apache.felix.configadmin;version='[1.9.24,1.9.25)', org.apache.felix.http.servlet-api;version='[2.1.0,2.1.1)', org.apache.felix.scr;version='[2.2.2,2.2.3)', org.eclipse.emf.common;version='[2.28.0,2.28.1)', org.eclipse.emf.ecore;version='[2.33.0,2.33.1)', org.eclipse.emf.ecore.xmi;version='[2.18.0,2.18.1)', org.eclipse.jetty.alpn.client;version='[11.0.13,11.0.14)', org.eclipse.jetty.client;version='[11.0.13,11.0.14)', org.eclipse.jetty.http;version='[11.0.13,11.0.14)', org.eclipse.jetty.io;version='[11.0.13,11.0.14)', org.eclipse.jetty.security;version='[11.0.13,11.0.14)', org.eclipse.jetty.server;version='[11.0.13,11.0.14)', org.eclipse.jetty.util;version='[11.0.13,11.0.14)', org.eclipse.sensinact.gateway.core.annotation;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.api;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.emf-api;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.geo-json;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.impl;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.models.metadata;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.models.provider;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.device-factory.device-factory-core;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.device-factory.parser-csv;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.http.http-device-factory;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.http.http-device-factory-tests;version='[0.0.2,0.0.3)', org.gecko.emf.osgi.component;version='[5.0.0,5.0.1)', org.opentest4j;version='[1.2.0,1.2.1)', org.osgi.service.component;version='[1.5.0,1.5.1)', org.osgi.service.typedevent;version='[1.0.0,1.0.1)', org.osgi.test.common;version='[1.2.1,1.2.2)', org.osgi.test.junit5;version='[1.2.1,1.2.2)', org.osgi.util.converter;version='[1.0.9,1.0.10)', org.osgi.util.function;version='[1.1.0,1.1.1)', org.osgi.util.promise;version='[1.3.0,1.3.1)', org.osgi.util.pushstream;version='[1.0.2,1.0.3)', slf4j.api;version='[1.7.36,1.7.37)', slf4j.simple;version='[1.7.36,1.7.37)']
```

It would be nice, if we can provide the delta as well, as different systems sometimes produce different results.

**Comment by juergen-albert** [comment]
Here it is fairly readable, as I just realized. In most consoles it looks more like this:

<img width="482" alt="image" src="https://github.com/bndtools/bnd/assets/8199490/24cf643f-7226-46cf-b605-1a923b0c4963">

**Comment by juergen-albert** [comment]
@pkriens I'm willing to provide a PR, but I can't say when at the moment.

**Comment by chrisrueger** [comment]
@juergen-albert I tried a fix in PR #6027 
Let me know if this is what you had in mind. 

It would print an additional line like this:

```
Diff [osgi.enroute.junit.wrapper;version='[4.12.0,4.12.1)', test.simple;version=snapshot] exist in calculated runbundles but missing in existing runbundles
```

Based on your example it would print an additional line: 

```
Error:  Error   :    Diff [org.osgi.service.cm;version='[1.6.1,1.6.2)'] exist in existing runbundles but missing in calculated runbundles
```

like

```
Error:  Error   :    Existing runbundles [com.fasterxml.jackson.core.jackson-annotations;version='[2.16.1,2.16.2)', com.fasterxml.jackson.core.jackson-core;version='[2.16.1,2.16.2)', com.fasterxml.jackson.core.jackson-databind;version='[2.16.1,2.16.2)', io.dropwizard.metrics.core;version='[4.2.19,4.2.20)', junit-jupiter-api;version='[5.9.0,5.9.1)', junit-jupiter-engine;version='[5.9.0,5.9.1)', junit-jupiter-params;version='[5.9.0,5.9.1)', junit-platform-commons;version='[1.9.0,1.9.1)', junit-platform-engine;version='[1.9.0,1.9.1)', junit-platform-launcher;version='[1.9.0,1.9.1)', org.apache.aries.component-dsl.component-dsl;version='[1.2.2,1.2.3)', org.apache.aries.typedevent.bus;version='[0.0.2,0.0.3)', org.apache.commons.commons-csv;version='[1.9.0,1.9.1)', org.apache.felix.configadmin;version='[1.9.24,1.9.25)', org.apache.felix.http.servlet-api;version='[2.1.0,2.1.1)', org.apache.felix.scr;version='[2.2.2,2.2.3)', org.eclipse.emf.common;version='[2.28.0,2.28.1)', org.eclipse.emf.ecore;version='[2.33.0,2.33.1)', org.eclipse.emf.ecore.xmi;version='[2.18.0,2.18.1)', org.eclipse.jetty.alpn.client;version='[11.0.13,11.0.14)', org.eclipse.jetty.client;version='[11.0.13,11.0.14)', org.eclipse.jetty.http;version='[11.0.13,11.0.14)', org.eclipse.jetty.io;version='[11.0.13,11.0.14)', org.eclipse.jetty.security;version='[11.0.13,11.0.14)', org.eclipse.jetty.server;version='[11.0.13,11.0.14)', org.eclipse.jetty.util;version='[11.0.13,11.0.14)', org.eclipse.sensinact.gateway.core.annotation;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.api;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.emf-api;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.geo-json;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.impl;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.models.metadata;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.models.provider;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.device-factory.device-factory-core;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.device-factory.parser-csv;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.http.http-device-factory;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.http.http-device-factory-tests;version='[0.0.2,0.0.3)', org.gecko.emf.osgi.component;version='[5.0.0,5.0.1)', org.opentest4j;version='[1.2.0,1.2.1)', org.osgi.service.cm;version='[1.6.1,1.6.2)', org.osgi.service.component;version='[1.5.0,1.5.1)', org.osgi.service.typedevent;version='[1.0.0,1.0.1)', org.osgi.test.common;version='[1.2.1,1.2.2)', org.osgi.test.junit5;version='[1.2.1,1.2.2)', org.osgi.util.converter;version='[1.0.9,1.0.10)', org.osgi.util.function;version='[1.1.0,1.1.1)', org.osgi.util.promise;version='[1.3.0,1.3.1)', org.osgi.util.pushstream;version='[1.0.2,1.0.3)', slf4j.api;version='[1.7.36,1.7.37)', slf4j.simple;version='[1.7.36,1.7.37)']
Error:  Error   :    Calculated runbundles [com.fasterxml.jackson.core.jackson-annotations;version='[2.16.1,2.16.2)', com.fasterxml.jackson.core.jackson-core;version='[2.16.1,2.16.2)', com.fasterxml.jackson.core.jackson-databind;version='[2.16.1,2.16.2)', io.dropwizard.metrics.core;version='[4.2.19,4.2.20)', junit-jupiter-api;version='[5.9.0,5.9.1)', junit-jupiter-engine;version='[5.9.0,5.9.1)', junit-jupiter-params;version='[5.9.0,5.9.1)', junit-platform-commons;version='[1.9.0,1.9.1)', junit-platform-engine;version='[1.9.0,1.9.1)', junit-platform-launcher;version='[1.9.0,1.9.1)', org.apache.aries.component-dsl.component-dsl;version='[1.2.2,1.2.3)', org.apache.aries.typedevent.bus;version='[0.0.2,0.0.3)', org.apache.commons.commons-csv;version='[1.9.0,1.9.1)', org.apache.felix.configadmin;version='[1.9.24,1.9.25)', org.apache.felix.http.servlet-api;version='[2.1.0,2.1.1)', org.apache.felix.scr;version='[2.2.2,2.2.3)', org.eclipse.emf.common;version='[2.28.0,2.28.1)', org.eclipse.emf.ecore;version='[2.33.0,2.33.1)', org.eclipse.emf.ecore.xmi;version='[2.18.0,2.18.1)', org.eclipse.jetty.alpn.client;version='[11.0.13,11.0.14)', org.eclipse.jetty.client;version='[11.0.13,11.0.14)', org.eclipse.jetty.http;version='[11.0.13,11.0.14)', org.eclipse.jetty.io;version='[11.0.13,11.0.14)', org.eclipse.jetty.security;version='[11.0.13,11.0.14)', org.eclipse.jetty.server;version='[11.0.13,11.0.14)', org.eclipse.jetty.util;version='[11.0.13,11.0.14)', org.eclipse.sensinact.gateway.core.annotation;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.api;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.emf-api;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.geo-json;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.impl;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.models.metadata;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.core.models.provider;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.device-factory.device-factory-core;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.device-factory.parser-csv;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.http.http-device-factory;version='[0.0.2,0.0.3)', org.eclipse.sensinact.gateway.southbound.http.http-device-factory-tests;version='[0.0.2,0.0.3)', org.gecko.emf.osgi.component;version='[5.0.0,5.0.1)', org.opentest4j;version='[1.2.0,1.2.1)', org.osgi.service.component;version='[1.5.0,1.5.1)', org.osgi.service.typedevent;version='[1.0.0,1.0.1)', org.osgi.test.common;version='[1.2.1,1.2.2)', org.osgi.test.junit5;version='[1.2.1,1.2.2)', org.osgi.util.converter;version='[1.0.9,1.0.10)', org.osgi.util.function;version='[1.1.0,1.1.1)', org.osgi.util.promise;version='[1.3.0,1.3.1)', org.osgi.util.pushstream;version='[1.0.2,1.0.3)', slf4j.api;version='[1.7.36,1.7.37)', slf4j.simple;version='[1.7.36,1.7.37)']
Error:  Error   :    Diff [org.osgi.service.cm;version='[1.6.1,1.6.2)'] exist in existing runbundles but missing in calculated runbundles

```
